lib/test_formulae.rb

```diff
@@ -132,38 +134,47 @@ def download_artifact_from_previous_run!(artifact_name, dry_run:)
         check_suite_nodes = response.dig("repository", "object", "checkSuites", "nodes")
         return if check_suite_nodes.blank?
 
-        wanted_artifact = artifact_metadata(check_suite_nodes, github_repository, "pull_request",
-                                            "CI", "conclusion", artifact_name)
-        # If we didn't find the artifact that we wanted, fall back to the `event_payload` artifact.
-        wanted_artifact ||= artifact_metadata(check_suite_nodes, github_repository, "pull_request_target",
-                                              "Triage tasks", "upload-metadata", "event_payload")
-        return if wanted_artifact.blank?
+        wanted_artifacts = artifact_metadata(check_suite_nodes, github_repository, "pull_request",
+                                             "CI", "conclusion", artifact_pattern)
+        wanted_artifacts_pattern = artifact_pattern
+        if wanted_artifacts.empty?
+          # If we didn't find the artifacts that we wanted, fall back to the `event_payload` artifact.
+          wanted_artifacts = artifact_metadata(check_suite_nodes, github_repository, "pull_request_target",
+                                               "Triage tasks", "upload-metadata", "event_payload")
+          wanted_artifacts_pattern = "event_payload"
+        end
+        return if wanted_artifacts.empty?
 
-        wanted_artifact_name = wanted_artifact.fetch("name")
-        if @downloaded_artifacts[sha].include?(wanted_artifact_name)
-          opoo "Already tried #{wanted_artifact_name} from #{sha}, giving up"
+        if (attempted_artifact = wanted_artifacts.find do |artifact|
+              @downloaded_artifacts[sha].include?(artifact.fetch("name"))
+            end)
+          opoo "Already tried #{attempted_artifact.fetch("name")} from #{sha}, giving up"
```

Comment:
```suggestion
        if wanted_artifacts.all? do |artifact|
              @downloaded_artifacts[sha].include?(artifact.fetch("name"))
            end
          opoo "Already tried #{wanted_artifacts_pattern} from #{sha}, giving up"
```
Maybe? Will need to think a bit about it though.

Reply:
Currently this is only used for `bottles`, `dependents` and `event_payload`, so the essentially difference here is whether to fail when e.g., `bottles-12` has been downloaded from the same commit but `bottles-13` has not.

My understanding is that this shouldn't usually happen, because `GitHub.download_artifact` below should have already thrown an error earlier (which we don't catch). So, I'm inclined to failing here for a single repeated attempt in the unlikely case that something funky happens. We'd only lose the cache anyway.